gui/wxpython/mapdisp/settings.py

```diff
@@ -0,0 +1,305 @@
+"""
+@package mapdisp.settings
+
+@brief Classes for map display settings management
+
+Classes:
+ - settings::ChBRender
+ - settings::ChBShowRegion
+ - settings::ChBAlignExtent
+ - settings::ChBResolution
+ - settings::MapDisplayPreferencesDialog
+
+(C) 2021 by the GRASS Development Team
+
+This program is free software under the GNU General Public License
+(>=v2). Read the file COPYING that comes with GRASS for details.
+
+@author Vaclav Petras <wenzeslaus gmail.com>
+@author Anna Kratochvilova <kratochanna gmail.com>
+@author Linda Kladivova <lindakladivova gmail.com>
+"""
+
+import wx
+import wx.lib.scrolledpanel as SP
+from gui_core.wrap import StaticBox
+
+from core.settings import UserSettings
+from gui_core.preferences import PreferencesBaseDialog
+
+
+class ChBItem:
+    """Base class for checkbox settings items"""
+
+    def __init__(self, parent, mapframe):

```

Comment:
Ideally, they shouldn't know about mapFrame and use signals or so.

Reply:
Ok, I've incorporated giface..